### PR TITLE
refactor: Port MPILogger to Swift

### PR DESCRIPTION
**Pull request by BrandonStalnaker**
## Summary
 - Port MPILogger to Swift. Do not replace existing implementation as Swift version can only be used in Swift.

 ## Testing Plan
 - Tested on sample app in simulator. Confirmed all unit tests as well

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-6476

### CODE REVIEW

File: mParticle-Apple-SDK/Logger/MPLogger.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+public struct MPLogger {
+    
+    public static func MPLogger(loggerLevel: MPILogLevel, format: String, arguments: any CVarArg...) {
+        if (MParticle.sharedInstance().logLevel.rawValue >= loggerLevel.rawValue && loggerLevel != .none) {
+            let msg = String.localizedStringWithFormat(format, arguments)
+// Custom Logger is marked as nonnull despite the fact that we expect it to be null if not set by the client
+//         if let customLogger = MParticle.sharedInstance().customLogger {
+//             customLogger(msg)
+//         } else
```
Comment (einsteinx2) [inline mParticle-Apple-SDK/Logger/MPLogger.swift:18]:
```suggestion
```
I assume this was meant to be deleted?

File: mParticle-Apple-SDK/Logger/MPLogger.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+//  mParticle-Apple-SDK
+//
+//  Created by Brandon Stalnaker on 9/22/24.
+//
+
+import Foundation
+
+public struct MPLogger {
+    
+    public static func MPLogger(loggerLevel: MPILogLevel, format: String, arguments: any CVarArg...) {
```
Comment (einsteinx2) [inline mParticle-Apple-SDK/Logger/MPLogger.swift:12]: Just a suggestion, but maybe we should mark this private to force the use of the other methods below. There should never be a need to call this one directly

**Review by einsteinx2 — CHANGES_REQUESTED** [review]

File: mParticle-Apple-SDK/Logger/MPLogger.swift (hunk 2 above)
Comment (BrandonStalnaker) [reply mParticle-Apple-SDK/Logger/MPLogger.swift:12]: that makes sense to me

File: mParticle-Apple-SDK/Logger/MPLogger.swift (hunk 1 above)
Comment (BrandonStalnaker) [reply mParticle-Apple-SDK/Logger/MPLogger.swift:18]: no I wanted to leave this here to make sure we understood this is kinda a todo once we fully convert to Swift

**Comment by BrandonStalnaker** [comment]
Addressed comments ready for a new look @einsteinx2

**Review by einsteinx2 — APPROVED** [review]
Changed nullability in public interface per my request. LGTM

**Pull request merged**